### PR TITLE
Fixes #2306 - remove unused CSS

### LINKED ISSUE #2306: remove unsused CSS
- [x] add pkg, grunt task & npm script https://www.npmjs.com/package/grunt-purifycss
- [x] remove unused CSS 
- [x] check for `webcompat-reset.css`
- [x] check `color.css`
- [x] check `styleguide.css`

### PR DESCRIPTION
**Pull request by zoepage**
r? @miketaylr

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -29,7 +29,8 @@ module.exports = function(grunt) {
     "concat",
     "uglify",
     "postcss",
-    "cssmin"
+    "cssmin",
+    "purifycss"
   ]);
```
Comment (miketaylr) [inline Gruntfile.js]: Can we also add it to the deploy task please?

File: grunt-tasks/purifycss.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+module.exports = function(grunt) {
+  grunt.config("purifycss", {
+    options: {
+      rejected: true,
+      info: true
+    },
+    target: {
+      src: [
+        "webcompat/templates/*.html",
+        "webcompat/templates/**/*.html",
```
Comment (miketaylr) [inline grunt-tasks/purifycss.js]: `**` matches zero or more directories (when its the only thing in the path), so the first line is redundant. Just `"webcompat/templates/**/*.html"` should be sufficient.

File: grunt-tasks/purifycss.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+module.exports = function(grunt) {
+  grunt.config("purifycss", {
+    options: {
+      rejected: true,
```
Comment (miketaylr) [inline grunt-tasks/purifycss.js]: So the idea is we create a stylesheet that *only* contains the unused styles? And then presumably we send in PRs from time to time removing them?

Could we add something somewhere in docs to explain this?

Comment (miketaylr) [reply grunt-tasks/purifycss.js]: (if we're crunched on time, filing a follow up bug to document is this is also cool)

File: Gruntfile.js (hunk 1 above)
Comment (zoepage) [reply Gruntfile.js]: Right now we are generating a CSS which is stripped, which we also can use as our `*.min.css` version, but I am not fully trusting that one yet. 

The purify task is showing the not used classes right now.

File: grunt-tasks/purifycss.js (hunk 3 above)
Comment (zoepage) [reply grunt-tasks/purifycss.js]: No, we create one without the removed styles. The `rejected` option is showing in the logs the unsused styles that have been removed.

Comment (miketaylr) [reply grunt-tasks/purifycss.js]: Ah, thanks!

**Pull request merged**